src/core/dependencies/parser/CircomFilesParser.ts

```diff
@@ -99,7 +101,7 @@ export class CircomFilesParser {
     const circomTemplateInputsVisitor = new CircomTemplateInputsVisitor(
       circomResolvedFile.absolutePath,
       circomResolvedFile.fileData.parsedFileData.templates[templateName].context,
-      parameterValues,
+      cloneDeep(parameterValues),
```

Comment:
Why do we need to copy?

Reply:
Inside the `CircomTemplateInputsVisitor`, information about the various var variables that are inside the template is written to this Record. Without copying this information will be saved and the parameters field will contain not only parameters of the main component, but also internal information

Reply:
But doesn't `cloneDeep` do a full deep copy of all the records, including the nested records?